angr/analyses/decompiler/region_simplifier.py

```diff
@@ -269,6 +269,71 @@ def _is_statement_terminating(self, stmt):
         return False
 
 
+class CascadingIfsRemover(SequenceWalker):
+    """
+    Coalesce cascading If constructs. Transforming the following construct::
+
+        if (cond_a) {
+            if (cond_b) {
+                true_body
+            } else { }
+        } else { }
+
+    into::
+
+        if (cond_a and cond_b) {
+            true_body
+        } else { }
+    """
+    def __init__(self, node):
+        handlers = {
+            SequenceNode: self._handle_Sequence,
+            CodeNode: self._handle_Code,
+            MultiNode: self._handle_MultiNode,
+            LoopNode: self._handle_Loop,
+            ConditionNode: self._handle_Condition,
+        }
+
+        super().__init__(handlers)
+        self.walk(node)
+
+    def _handle_Condition(self, node, parent=None, index=None, **kwargs):
+        """
+
+        :param ConditionNode node:
+        :param successor:
+        :return:
+        """
+
+        if node.true_node is not None:
+            self._handle(node.true_node, parent=node, index=0)
+        if node.false_node is not None:
+            self._handle(node.false_node, parent=node, index=1)
+
+        if node.true_node is not None and node.false_node is None:
+            if isinstance(node.true_node, SequenceNode):
+                last_node = None
+                if len(node.true_node.nodes) > 1 and all(self.is_empty_node(node_) for node_ in node.true_node.nodes[:-1]):
+                    last_node = node.true_node.nodes[-1]
+                elif len(node.true_node.nodes) == 1:
+                    last_node = node.true_node.nodes[0]
+
+                true_node = last_node
+
+                if isinstance(true_node, ConditionNode) and true_node.true_node is not None and true_node.false_node is None:
```

Comment:
Is it possible for these to be swapped? With `true_node.true_node` being `None` and `true_node.false_node` being `not None`? Or will `true_node` always be populated first, in this case with a boolean complement?

Reply:
Good catch. There is another rule that converts `if (a) { } else { something }` to `if(!a) {something}` already.